block.go

```diff
@@ -124,7 +124,7 @@ func CompressBlock(src, dst []byte, hashTable []int) (_ int, err error) {
 		si, mLen = si+mLen, si+minMatch
 
 		// Find the longest match by looking by batches of 8 bytes.
-		for si < sn {
+		for sn-si >= 8 {
```

Comment:
Should actually be:
for si+8 < sn {

Reply:
Sorry for the long delay, holidays and all...

I've updated the condition. It does not seem to have any significant impact in this case. 

Did you see the comment in https://github.com/python-lz4/python-lz4/issues/183? Do you know how it compares to compressing it with the C-library directly?